mod-data-export/src/test/resources/domain/dataexport/features/export.feature

```diff
@@ -0,0 +1,144 @@
+Feature: Tests for uploading "uuids file" and exporting the records
+
+  Background:
+    * url baseUrl
+
+    * callonce login testAdmin
+    * def okapiAdminToken = okapitoken
+
+    * callonce login testUser
+    * def okapiUserToken = okapitoken
+
+    * callonce loadTestVariables
+
+    * def headersUser = { 'Content-Type': 'application/json', 'x-okapi-token': '#(okapiUserToken)', 'Accept': 'application/json'  }
+    * def headersUserOctetStream = { 'Content-Type': 'application/octet-stream', 'x-okapi-token': '#(okapiUserToken)', 'Accept': 'application/json'  }
+    * configure headers = headersUser
+
+  #Positive scenarios
+
+  Scenario Outline: test upload file and export flow.
+    #should create file definition
+    Given path 'data-export/file-definitions'
+    And def fileDefinition = {'id':'#(<fileDefinitionId>)','fileName':'<fileName>', 'uploadFormat':'<uploadFormat>'}
+    And request fileDefinition
+    When method POST
+    Then status 201
+    And match response.status == 'NEW'
+    And match response.uploadFormat == '<uploadFormat>'
+
+    #should return created file definition
+    Given path 'data-export/file-definitions', <fileDefinitionId>
+    When method GET
+    Then status 200
+    And match response.status == 'NEW'
+    And match response.uploadFormat == '<uploadFormat>'
+
+    #should upload file by created file definition id
+    Given path 'data-export/file-definitions/',<fileDefinitionId>,'/upload'
+    And configure headers = headersUserOctetStream
+    And request karate.readAsString('classpath:samples/file-definition/<fileName>')
+    When method POST
+    Then status 200
+    And match response.jobExecutionId == '#present'
+    And match response.status == 'COMPLETED'
+    And match response.uploadFormat == '<uploadFormat>'
+    And match response.sourcePath == '#present'
+    And def jobExecutionId = response.jobExecutionId
+    And call pause 1000
```

Comment:
Could you please add a comment or name the step properly to understand why do we need the pause?

Is it pause of the test or some module pause?

Reply:
The problem here that when we upload a file it returns 204 immediately and starts the processing file in the background and the next API call requires this file to be uploaded but it may need some time to be done. That is why we need to wait a bit, but a better way is to use the 'retry until' condition by which we can check that the related to uploading file job has status 'completed', if it doesn't then we wait 500 ms and we make a call again to be sure that the job has completed status before invoking further calls that depend from the status of this job. So, I plugged in the "retry until" statement. As well I updated the description regarding these delays.